os/fs/driver/mtd/smart.c

```diff
@@ -1005,6 +1033,48 @@ static int smart_setsectorsize(FAR struct smart_struct_s *dev, uint16_t size)
 			dev->availSectPerBlk = dev->sectorsPerBlk;
 		}
 	}
+	
+#ifdef CONFIG_MTD_SMART_JOURNALING
+	/** Journal Sector is reserved at the last of smartfs partition, it doesn't use MTD Header.
+	  * Journal sectors divided to 2 area, this will be toggled if anyone is filled with data,
+	  * erase previous one. So total number of journal sector must be N*Eraseblock
+	  * We will use it as a contigous memory space...
+	  */
+	size_t log_size = sizeof(struct smart_journal_logging_data_s);
+	/** First get ideal size of journal sector. Formula is Below (x : data sectors y : journal sectors)
+	  * 1. dev->sectorsize : dev->njournalsectors = x : y  
```

Comment:
Could you please explain how this ratio is derived?

This ratio seems mathematically incorrect - 
Mathematically: **dev->sectorsize / dev->njournalsectors** would represent **bytes / sector**
                          **x / y** = data sectors / journal sectors represents **sector / sector**
                           But bytes/sector != sector/sector, so this ratio seems incorrect.

I think the number of journal sectors should depend on the ratio of **log_size/dev->sectorsize**
In addition, the number of journal entries that might be created for every application request will matter.

Reply:
its a typo error please see formula below of comments.
	uint16_t nsector = (dev->geo.neraseblocks * dev->sectorsPerBlk * log_size) / (dev->sectorsize + log_size);

formula #1 
512 : 13  = x : y
-> dev->sectorsize : log_size = x : y
-> x(log_size) = y(dev->sectorsize)
Formula #2
x + y = dev->geo.neraseblocks * dev->sectorsPerBlk
x = (dev->geo.neraseblocks * dev->sectorsPerBlk) - y;
Now we merge it,
log_size((dev->geo.neraseblocks * dev->sectorsPerBlk) - y) = (dev->sectorsize)y
-> (dev->geo.neraseblocks * dev->sectorsPerBlk) * log_size = (dev->sectorsize)y + (log_size) y
-> y = (dev->geo.neraseblocks * dev->sectorsPerBlk) * log_size) / (dev->sectorsize + log_size)

Reply:
The ratio seems mathematically valid now.
However, it only depends on the ratio of the sizes of the sectors and journal.
No dependency on the frequency and number of journals that might be written?
However, that would be tough to estimate.

Reply:
Still, formula #1 is an assumption made by us or is it a mathematically derived ratio?